client/reader/stats.js

```diff
@@ -99,6 +99,23 @@ function getLocation( path ) {
 	return 'unknown';
 }
 
+/**
+ *
+ * @param {Object} eventProperties extra event properties to add
+ * @param {*} pathnameOverride Overwrites location used for determining ui_algo. See notes in
+ * `recordTrack` function docs below for more info.
+ * @param {Object|null} post Optional post object used to build post event props.
+ * @returns new eventProperties object with default reader values added.
+ */
+export function buildReaderTracksEventProps( eventProperties, pathnameOverride, post ) {
+	const location = getLocation( pathnameOverride || window.location.pathname );
+	eventProperties = Object.assign( { ui_algo: location }, eventProperties );
+	if ( post ) {
+		eventProperties = Object.assign( getTracksPropertiesForPost( post ), eventProperties );
+	}
+	return eventProperties;
```

Comment:
Nit: I would prefer us using a new variable for the output here so it's a bit clearer that we're operating on a new/distinct "thing".
```suggestion
	let composedProperties = Object.assign( { ui_algo: location }, eventProperties );
	if ( post ) {
		composedProperties = Object.assign( getTracksPropertiesForPost( post ), composedProperties );
	}
	return composedProperties;
```